controllers/openshift_controller.go

```diff
@@ -846,72 +893,69 @@ func (r *KataConfigOpenShiftReconciler) processKataConfigDeleteRequest() (ctrl.R
 			r.Log.Error(err, "Error found deleting machine config. If the machine config exists after installation it can be safely deleted manually.",
 				"mc", mc.Name)
 		}
-		// Sleep for MCP to reflect the changes
-		r.Log.Info("Pausing for a minute to make sure mcp has started syncing up")
-		time.Sleep(60 * time.Second)
+		r.Log.Info("Starting to wait for MCO to start reconciliation")
+		r.kataConfig.Status.WaitingForMcoToStart = true
```

Comment:
I know this is not part of this PR, but when we added the new field `WaitingForMcoToStart` to the KataConfig CRD we changed the API. Did we increase the version of KataConfig and/or did we test upgrading the cluster?

Reply:
No and no (well, that's pending but no so far afaik).  We handle it the same way as we did `KataConfig.spec.logLevel` addition or `KataConfig.spec.kataMonitorImage` removal.  From the docs I'm under the impression that once we start increasing version we have to go all the way through mutating webhooks to the heavyweight machinery necessary to do it right.  I'm further under the impression that we're still rather far from the point where this would start making sense and being worth the trouble but I might be wrong here, with first actual deployments underway.

Reply:
We'll start testing upgrade cases soon and fix what needs to be fixed.